openedx/frontend-lib-content-components
# Re-write the dropzone component and fix styling issues

## Description

We are planning to use the Dropzone Paragone component here and also fix the style for URL video uploader. We are trying to keep the paragon component intact for it to be used.

Introduced new thumbnail

![image](https://github.com/openedx/frontend-lib-content-components/assets/7670449/bb776db6-4ca2-47ea-b500-d91ed1f6e13f)

Introduced URL uploader field

![image](https://github.com/openedx/frontend-lib-content-components/assets/7670449/a0a2aeca-b774-43dd-b992-9fe4b7e9dfc5)


We remove the custom Dropzone that was written initially and have used the paragon inbuilt one.


## Supporting information



## Testing instructions

1. Set up the master devstack, make sure you have course-authoring-mfe(frontend-app-course-authoring) running
2. We have to clone this repo in {EDX_DEVSTACK}/src directory.
3. Change the branch being used to `open-craft:farhaan/fix-drag-drop-component` and run `make build`
4. Follow the steps mentioned in https://github.com/openedx/frontend-lib-content-components/pull/262 to set up video editor, make sure to `Step 3 - Step 7`
5. We need to upload a small video and connect it to the demo course as mentioned in https://github.com/openedx/frontend-lib-content-components/pull/326
6. We need to make changes to the edx-platform to dummy S3 connection as mentioned in the above PR.
7. Now add a video xblock in any unit, you will be redirected to the the course-authoring-mfe's video setting page.
8. Check the thumbnail here -- new addition
8. Go to "Replace Video"
9. Try to upload a video, you should see a spinner here.
10. You will see the new URL bar here as well -- this works like a search field.


## Deadline


## Other information



Private ref: https://tasks.opencraft.com/browse/BB-7524

Reviewer:
